### DISCUSSION TITLE
Support OpenRouter

### DISCUSSION THREAD

**Discussion by CyborgArmy83** [Taewan-P/gpt_mobile#107] [Ideas]
OpenRouter has a reputable name in our community and is a very nice way to run almost all big brand models from one API. I think that many people would benefit from officially implementing their API, this will be quite easy since it's closely mimics other known API's but I believe it does offer some more features such as selecting models remotely.

**Comment by Taewan-P** [comment]
Hi! For a temporary workaround, https://github.com/Taewan-P/gpt_mobile/discussions/62#discussioncomment-10928255 will work. I will consider this for the next update..

**Comment by muntedcrocodile** [comment]
Yes please. Would be great if u could also specify some additional custom data for the request so openrouters routing settings can be configured.

**Reply by Taewan-P** [reply]
Can you tell me more about the custom data? What routing settings to be exact?

**Comment by ThoughtPhotography** [comment]
I found this app through a DeepSearch with Grok 3 asking for Android apps that support OpenRouter as an endpoint.
The workaround above works but a dedicated setting for OpenRouter could get the full list of models (similar to Open WebUI) to choose from.

Ideally, when using OpenRouter we could pick the model in the chat UI (vs editing the settings all the time). The full list is massive so most users would want to narrow it to their favorites.

OpenRouter is really the best solution IMO, as one account/API key is better than half a dozen, and most users would prefer a US hosted version of DeepSeek.

**Comment by Taewan-P** [comment]
Hello, this feature is now implemented in [v0.7.0](https://github.com/Taewan-P/gpt_mobile/releases). Thanks!
